src/components/PostDetails.tsx

```diff
@@ -1,117 +1,144 @@
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../api/services';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [isOpenedForm, setIsOpenedForm] = useState(false);
+
+  const onSetError = () => {
+    setError(true);
+  };
+
+  const loadCommentsFromServer = async (post: Post) => {
+    setLoading(true);
+
+    try {
+      const usersFromServer = await getComments(post.id);
+
+      setComments(usersFromServer);
+    } catch {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const addNewComment = (newComment: Comment) => {
+    setComments(prev => [...prev, newComment]);
+  };
+
+  useEffect(() => {
+    if (selectedPost) {
+      setError(false);
+      setComments([]);
+      loadCommentsFromServer(selectedPost);
+    }
+  }, [selectedPost]);
+
+  const handleDeleteComment = (id: number) => {
+    try {
+      deleteComment(id);
+
+      setComments((prev) => prev.filter(com => com.id !== id));
+    } catch {
+      setComments(prev => prev);
+    }
+  };
+
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${selectedPost?.id}: ${selectedPost?.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {`${selectedPost?.body}`}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
+          {loading && <Loader />}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+          {error
+            && (
+              <div className="notification is-danger" data-cy="CommentsError">
+                Something went wrong
+              </div>
+            )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {comments?.length === 0 && !loading
+            && (
+              <p className="title is-4" data-cy="NoCommentsMessage">
+                No comments yet
+              </p>
+            )}
+
+          {comments?.length !== 0 && !loading
+            && (
+              <>
+                <p className="title is-4">Comments:</p>
+                {comments?.map(comment => {
+                  return (
```

Comment:
```suggestion
                {comments?.map(comment => (
```

can be more simplify